controllers/packetmachine_controller.go

```diff
@@ -195,28 +191,54 @@ func (r *PacketMachineReconciler) reconcile(ctx context.Context, machineScope *s
 	}
 
 	providerID := machineScope.GetInstanceID()
+	defaultTags := defaultTags(machineScope.Namespace(), machineScope.Name(), clusterScope.Name())
+
 	var (
 		dev                  *packngo.Device
 		addrs                []corev1.NodeAddress
 		err                  error
 		controlPlaneEndpoint packngo.IPAddressReservation
 	)
-	// if we have no provider ID, then we are creating
+
 	if providerID != "" {
+		// If we already have a providerID, then retrieve the device using the
+		// providerID. This means that the Machine has already been created
+		// and we successfully recorded the providerID.
 		dev, err = r.PacketClient.GetDevice(providerID)
+		if err != nil {
+			var perr *packngo.ErrorResponse
+			if errors.As(err, &perr) && perr.Response != nil && perr.Response.StatusCode == http.StatusNotFound {
```

Comment:
Perhaps we should also capture `http.StatusForbidden` for #245, unless the behavior has changed in the provider API.